src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisParametersService.java

```diff
@@ -0,0 +1,147 @@
+        Optional<SecurityAnalysisParametersEntity> securityAnalysisParametersEntity = Optional.empty();
+        if (parametersUuid != null) {
+            securityAnalysisParametersEntity = securityAnalysisParametersRepository.findById(parametersUuid);
+        }
+        SecurityAnalysisParameters parameters = toSecurityAnalysisParameters(securityAnalysisParametersEntity.orElse(null));
+        return new SecurityAnalysisRunContext(
+                networkUuid,
+                variantId,
+                contingencyListNames,
+                receiver,
+                provider,
+                parameters,
+                loadFlowParametersInfos,
+                reportUuid,
+                reporterId,
+                reportType,
+                userId);
+
+    }
+
+    public static SecurityAnalysisParameters toSecurityAnalysisParameters(SecurityAnalysisParametersEntity entity) {
+        if (entity == null) {
+            return SecurityAnalysisParameters.load()
+                    // the default values are overloaded
+                    .setIncreasedViolationsParameters(getIncreasedViolationsParameters(DEFAULT_FLOW_PROPORTIONAL_THRESHOLD, DEFAULT_LOW_VOLTAGE_PROPORTIONAL_THRESHOLD, DEFAULT_LOW_VOLTAGE_ABSOLUTE_THRESHOLD, DEFAULT_HIGH_VOLTAGE_PROPORTIONAL_THRESHOLD, DEFAULT_HIGH_VOLTAGE_ABSOLUTE_THRESHOLD));
+        }
+        SecurityAnalysisParameters securityAnalysisParameters = new SecurityAnalysisParameters();
+        securityAnalysisParameters.setIncreasedViolationsParameters(getIncreasedViolationsParameters(entity.getFlowProportionalThreshold(), entity.getLowVoltageProportionalThreshold(), entity.getLowVoltageAbsoluteThreshold(), entity.getHighVoltageProportionalThreshold(), entity.getHighVoltageAbsoluteThreshold()));
+        return securityAnalysisParameters;
+    }
+
+    public static SecurityAnalysisParameters.IncreasedViolationsParameters getIncreasedViolationsParameters(double flowProportionalThreshold, double lowVoltageProportionalThreshold, double lowVoltageAbsoluteThreshold, double highVoltageProportionalThreshold, double highVoltageAbsoluteThreshold) {
+        SecurityAnalysisParameters.IncreasedViolationsParameters increasedViolationsParameters = new SecurityAnalysisParameters.IncreasedViolationsParameters();
+        increasedViolationsParameters.setFlowProportionalThreshold(flowProportionalThreshold);
+        increasedViolationsParameters.setLowVoltageAbsoluteThreshold(lowVoltageAbsoluteThreshold);
+        increasedViolationsParameters.setLowVoltageProportionalThreshold(lowVoltageProportionalThreshold);
+        increasedViolationsParameters.setHighVoltageAbsoluteThreshold(highVoltageAbsoluteThreshold);
+        increasedViolationsParameters.setHighVoltageProportionalThreshold(highVoltageProportionalThreshold);
+        return increasedViolationsParameters;
+    }
+
+    public static SecurityAnalysisParametersValues fromEntity(SecurityAnalysisParametersEntity entity) {
```

Comment:
Isn't this the same as `toSecurityAnalysisParametersValues` in `SecurityAnalysisParametersEntity` ?
If it's not the case, I suggest that you move it to the Entity's file